lib/src/google_fonts_base.dart

```diff
@@ -110,23 +114,33 @@ TextStyle googleFontsTextStyle({
 /// disk, then it fetches it via the [fontUrl], stores it on disk, and loads it
 /// into the [FontLoader].
 Future<void> loadFontIfNecessary(GoogleFontsDescriptor descriptor) async {
-  final familyWithVariant = descriptor.familyWithVariant();
+  final familyWithVariantString = descriptor.familyWithVariant.toString();
   // If this font has already been loaded, then there is no need to load it
   // again.
-  if (_loadedFonts.contains(familyWithVariant)) {
+  if (_loadedFonts.contains(familyWithVariantString)) {
+    return;
+  }
+
+  // If this font can be loaded by the pre-bundled assets, then there is no
+  // need to load it at all.
+  final assetManifestJson = await _loadAssetManifestJson();
+
+  if (_isFamilyWithVariantInManifest(
+      descriptor.familyWithVariant, assetManifestJson)) {
     return;
   }
 
-  _loadedFonts.add(familyWithVariant);
-  final fontLoader = FontLoader(familyWithVariant);
+  _loadedFonts.add(familyWithVariantString);
+  final fontLoader = FontLoader(familyWithVariantString);
 
   Future<ByteData> byteData;
   if (!kIsWeb) {
-    byteData = _readLocalFont(familyWithVariant);
+    byteData = _loadFontFromDeviceFileSystem(familyWithVariantString);
   }
   final localFontFound = byteData != null && await byteData != null;
   if (!localFontFound) {
-    byteData = _httpFetchFont(familyWithVariant, descriptor.fontUrl);
+    byteData = _httpFetchFontAndSaveToDevice(
+        familyWithVariantString, descriptor.fontUrl);
```

Comment:
nit: add trailing comma and reformat.

```suggestion
        familyWithVariantString, descriptor.fontUrl,);
```